scripts/ci/docker-test.sh

```diff
@@ -75,17 +75,37 @@ checkpoint_container () {
 	docker wait cr
 }
 
-restore_container () {
-	CHECKPOINT_NAME=$1
-
-	docker start --checkpoint "$CHECKPOINT_NAME" cr 2>&1 | tee log || {
+print_logs () {
 	cat "$(grep log 'log file:' | sed 's/log file:\s*//')" || true
 		docker logs cr || true
 		cat $CRIU_LOG || true
 		dmesg
 		docker ps
 		exit 1
-	}
+}
+
+declare -i max_restore_container_tries=3
+current_iteration=
+
+restore_container () {
+	CHECKPOINT_NAME=$1
+
+	docker start --checkpoint "$CHECKPOINT_NAME" cr 2>&1 | tee log || {
+		# FIXME: There is a race condition in docker/containerd that causes
+		# docker to occasionally fail when starting a container from a
+		# checkpoint immediately after the checkpoint has been created.
+		# https://github.com/moby/moby/issues/42900
+		if [ "$current_iteration" -gt "$max_restore_container_tries" ]; then
```

Comment:
https://github.com/checkpoint-restore/criu/actions/runs/6948976416/job/18929387032?pr=2299
```
+ '[' '' -gt 3 ']'
1800
./docker-test.sh: line 100: [: : integer expression expected
```

Reply:
Thanks for pointing this out. I opened a pull request to fix this: https://github.com/checkpoint-restore/criu/pull/2302